packages/teleport/src/Discover/SelectResource/SelectResource.tsx

```diff
@@ -35,22 +42,71 @@ import type { TabComponent } from 'design/SlideTabs/SlideTabs';
 import type { ResourceType, ResourceLocation } from '../resource-lists';
 import type { AgentStepProps } from '../types';
 import type { State } from '../useDiscover';
+import type { AuthType } from 'teleport/services/user';
 
 export default function Container(props: AgentStepProps) {
   const ctx = useTeleport();
-  return <SelectResource nextStep={props.nextStep} />;
+  const { clusterId, isLeafCluster } = useStickyClusterId();
+  const ctxState = ctx.storeUser.state;
+  return (
+    <SelectResource
+      authType={ctxState.authType}
+      canCreate={ctx.storeUser.getTokenAccess().create}
+      clusterId={clusterId}
+      isEnterprise={ctx.isEnterprise}
+      isLeafCluster={isLeafCluster}
```

Comment:
`isLeafCluster` required? this is only used to disable adding as users can only add new resources on root cluster